sdk/spring/spring-cloud-azure-service/src/test/java/com/azure/spring/service/implementation/eventhubs/EventProcessorClientBuilderFactoryTests.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.service.implementation.eventhubs;
+
+import com.azure.messaging.eventhubs.EventProcessorClientBuilder;
+import com.azure.spring.service.eventhubs.processor.EventProcessingListener;
+import com.azure.spring.service.implementation.AzureServiceClientBuilderFactoryBaseTests;
+import com.azure.spring.service.implementation.eventhubs.factory.EventProcessorClientBuilderFactory;
+import org.junit.jupiter.api.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class EventProcessorClientBuilderFactoryTests extends AzureServiceClientBuilderFactoryBaseTests<EventProcessorClientBuilder,
+    TestAzureEventHubsProperties, EventProcessorClientBuilderFactory> {
+
+    @Override
+    protected TestAzureEventHubsProperties createMinimalServiceProperties() {
+        return new TestAzureEventHubsProperties();
+    }
+
+    @Test
+    void customPrefetchCount() {
+        TestAzureEventHubsProperties properties = createMinimalServiceProperties();
+        properties.getProcessor().setPrefetchCount(150);
+        final TestEventProcessorClientBuilderFactory builderFactory = new TestEventProcessorClientBuilderFactory(properties);
```

Comment:
why not spy the target object directly?

Reply:
This method `prefetchCount` is our target to verify, then the builder `EventProcessorClientBuilder` should be a mock object. If using `spy` the factory class, then the real builder will be created.